homeassistant/components/yeelight/__init__.py

```diff
@@ -252,15 +262,33 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
 async def _async_setup_device(
     hass: HomeAssistant,
     host: str,
-    config: dict,
+    entry: ConfigEntry,
+    capabilities: Optional[dict],
 ) -> None:
+    # Get model from config and capabilities
+    model = entry.options.get(CONF_MODEL)
+    if not model and capabilities is not None:
+        model = capabilities.get("model")
+
     # Set up device
-    bulb = Bulb(host, model=config.get(CONF_MODEL) or None)
-    capabilities = await hass.async_add_executor_job(bulb.get_capabilities)
-    if capabilities is None:  # timeout
-        _LOGGER.error("Failed to get capabilities from %s", host)
-        raise ConfigEntryNotReady
-    device = YeelightDevice(hass, host, config, bulb)
+    bulb = Bulb(host, model=model or None)
+    if capabilities is None:
+        capabilities = await hass.async_add_executor_job(bulb.get_capabilities)
+
+    # Migrate name
+    if not entry.data.get(CONF_NAME):
+        # Disable update listener temporarily to avoid reload
+        hass.data[DOMAIN][DATA_CONFIG_ENTRIES][entry.entry_id][
+            DATA_UNSUB_UPDATE_LISTENER
+        ]()
+        _async_migrate_name(hass, entry, capabilities)
+        hass.data[DOMAIN][DATA_CONFIG_ENTRIES][entry.entry_id][
+            DATA_UNSUB_UPDATE_LISTENER
+        ] = entry.add_update_listener(_async_update_listener)
```

Comment:
It is not bad, it's just not necessary. Should I handle this? Maybe I can move this into `_async_migrate_name` and do the unsub/sub only when the migration actually happens. Or should I just remove that and reload the entry anyway?

Reply:
I would say move it to `_async_migrate_name` or remove it. It feels misplaced on this location.

Reply:
Hi I don't know if this is my set up problem or where to report this.
I have 3 yeelight lamps, 1 bedside lamp + 1 ceiling lamp. all of them are usually switched off and my family treats them as common lamps. Since 0.115 release (now I am on 0.115.5) if even one lamp is switched off I got the following in the configuration logs

`Log Details (WARNING)
Logger: homeassistant.config_entries
Source: config_entries.py:239
First occurred: 8:37:04 PM (16 occurrences)
Last logged: 8:42:33 PM

Config entry for yeelight not ready yet. Retrying in 5 seconds
Config entry for yeelight not ready yet. Retrying in 10 seconds
Config entry for yeelight not ready yet. Retrying in 20 seconds
Config entry for yeelight not ready yet. Retrying in 40 seconds
Config entry for yeelight not ready yet. Retrying in 80 seconds`

Is there something I could in order to avoid this behavior (without be forced to switch off the lamps from HA)?